interp/interp_test.go

```diff
@@ -329,58 +329,58 @@ var runTests = []runTest{
 	{`[[ $0 == "bash" || $0 == "gosh" ]]`, ""},
 
 	// dollar quotes
-	{`echo $'foo\nbar'`, "foo\nbar\n"},
+	{`echo $'foo_interp_does_not_exist\nbar_interp_does_not_exist'`, "foo_interp_does_not_exist\nbar_interp_does_not_exist\n"},
 	{`echo $'\r\t\\'`, "\r\t\\\n"},
-	{`echo $"foo\nbar"`, "foo\\nbar\n"},
+	{`echo $"foo_interp_does_not_exist\nbar_interp_does_not_exist"`, "foo_interp_does_not_exist\\nbar_interp_does_not_exist\n"},
 	{`echo $'%s'`, "%s\n"},
 	{`a=$'\r\t\\'; echo "$a"`, "\r\t\\\n"},
-	{`a=$"foo\nbar"; echo "$a"`, "foo\\nbar\n"},
+	{`a=$"foo_interp_does_not_exist\nbar_interp_does_not_exist"; echo "$a"`, "foo_interp_does_not_exist\\nbar_interp_does_not_exist\n"},
 	{`echo $'\a\b\e\E\f\v'`, "\a\b\x1b\x1b\f\v\n"},
 	{`echo $'\\\'\"\?'`, "\\'\"?\n"},
 	{`echo $'\1\45\12345\777\9'`, "\x01%S45\xff\\9\n"},
 	{`echo $'\x\xf\x09\xAB'`, "\\x\x0f\x09\xab\n"},
 	{`echo $'\u\uf\u09\uABCD\u00051234'`, "\\u\u000f\u0009\uabcd\u00051234\n"},
 	{`echo $'\U\Uf\U09\UABCD\U00051234'`, "\\U\u000f\u0009\uabcd\U00051234\n"},
 	{
-		"echo 'foo\x00bar'",
-		"foobar\n",
+		"echo 'foo_interp_does_not_exist\x00bar_interp_does_not_exist'",
+		"foo_interp_does_not_existbar_interp_does_not_exist\n",
 	},
 	{
-		"echo \"foo\x00bar\"",
-		"foobar\n",
+		"echo \"foo_interp_does_not_exist\x00bar_interp_does_not_exist\"",
+		"foo_interp_does_not_existbar_interp_does_not_exist\n",
 	},
 	{
-		"echo $'foo\x00bar'",
-		"foobar\n",
+		"echo $'foo_interp_does_not_exist\x00bar_interp_does_not_exist'",
+		"foo_interp_does_not_existbar_interp_does_not_exist\n",
 	},
 	{
-		"echo $'foo\\x00bar'",
-		"foo\n",
+		"echo $'foo_interp_does_not_exist\\x00bar_interp_does_not_exist'",
+		"foo_interp_does_not_exist\n",
 	},
 	{
-		"echo $'foo\\xbar'",
-		"foo\xbar\n",
+		"echo $'foo_interp_does_not_exist\\xbar_interp_does_not_exist'",
+		"foo_interp_does_not_exist\xbar_interp_does_not_exist\n",
```

Comment:
```suggestion
		"interp_does_not_exist\xinterp_does_not_exist_alt\n",
```

Note that such a form wouldn't work because `\xin` is a forbidden sequence (`\x must be followed by two hexadecimal digits`). `\xbar` is acceptable.

Hence we use `foo_interp_does_not_exist` rather than `interp_does_not_exist`

Reply:
Huh, I never intended hex escape codes to expand here. Can we use `\\x` in the second string as well?

Reply:
> Huh, I never intended hex escape codes to expand here. Can we use `\\x` in the second string as well?

Unfortunately not:

```
    interp_test.go:3242: wrong output in "echo $'foo_interp_does_not_exist\\xbar_interp_does_not_exist'":
        want: "foo_interp_does_not_exist\\xbar_interp_does_not_exist\n"
        got:  "foo_interp_does_not_exist\xbar_interp_does_not_exist\n"
```